json_serialization/reader.nim

```diff
@@ -449,6 +448,49 @@ func isBitwiseSubsetOf[N](lhs, rhs: array[N, uint]): bool =
 template isCharArray[N](v: array[N, char]): bool = true
 template isCharArray(v: auto): bool = false
 
+func parseStringEnum[T](
+    r: var JsonReader, value: var T,
+    stringNormalizer: static[proc(s: string): string]) =
+  try:
+    value = genEnumCaseStmt(
+      T, r.lexer.strVal,
+      default = nil, ord(T.low), ord(T.high), stringNormalizer)
+  except ValueError as err:
+    const typeName = typetraits.name(T)
+    r.raiseUnexpectedValue("Invalid value for '" & typeName & "'")
+
+func strictNormalize(s: string): string =  # Match enum value exactly
```

Comment:
I suspect the `genEnumCaseStmt` can work with a template as well and then the final code may be more efficient if we supply this identity function in the form of a template (that will be completely erased form the final code at compile-time).